toolbox/util/githubClient.go

```diff
@@ -164,10 +172,49 @@ func (g GithubClient) CloseFailedPullRequests(prTitlePrefix, repo, baseBranch st
 
 // GetHeadCommitSHA finds the SHA of the commit to which the HEAD of branch points
 func (g GithubClient) GetHeadCommitSHA(repo, branch string) (string, error) {
-	ref, _, err := g.client.Git.GetRef(
-		context.Background(), g.owner, repo, "refs/heads/"+branch)
+	return g.getReferenceSHA(repo, "refs/heads/"+branch)
+}
+
+// GetSHATime gets the time when sha is made
+func (g GithubClient) GetSHATime(repo, sha string) (*time.Time, error) {
+	commit, _, err := g.client.Git.GetCommit(
+		context.Background(), g.owner, repo, sha)
+	if err != nil {
+		return nil, err
+	}
+	return (*(*commit).Author).Date, nil
+}
+
+// GetTagPublishTime finds the time a tag is published
+func (g GithubClient) GetTagPublishTime(repo, tag string) (*time.Time, error) {
```

Comment:
Kind of suspicious, the time when a tag created is not necessarily to be the time when the corresponding SHA is created.  Could you double check the which time the api gives you. SHA time seems to have more value.